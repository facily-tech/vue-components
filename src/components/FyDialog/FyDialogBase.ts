/// <reference path="../../../node_modules/vuetify/src/globals.d.ts" />
import Vue from 'vue';

import { VDialog } from 'vuetify/lib';

const BaseSnackbar = Vue.extend({ extends: VDialog, mixins: [VDialog] });

type options = InstanceType<typeof BaseSnackbar>;

export default BaseSnackbar.extend<options>().extend({
  name: 'fy-dialog-base',
  computed: {
    class() {
      return {
        ...VDialog.options.computed.classes.call(this),
        'fy-dialog-base': true,
      };
    },
  },
});
