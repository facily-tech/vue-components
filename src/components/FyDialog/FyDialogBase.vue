<template>
  <v-dialog v-bind="$attrs" v-on="$listeners" />
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'FyDialogBase',
});
</script>
