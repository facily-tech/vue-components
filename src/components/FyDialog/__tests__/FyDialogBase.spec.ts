import Vue from 'vue';
import Vuetify from 'vuetify';
import FyDialogBase from '../FyDialogBase.vue';
import { mount, MountOptions, Wrapper } from '@vue/test-utils';

Vue.use(Vuetify);

describe('FyDialogBase.ts', () => {
  // eslint-disable-line max-statements
  type Instance = InstanceType<typeof FyDialogBase>;
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>;
  beforeEach(() => {
    mountFunction = (options?: MountOptions<Instance>) => {
      return mount(FyDialogBase, {
        vuetify: new Vuetify(),
        ...options,
      });
    };
  });

  it('should render component and match snapshot', () => {
    const wrapper = mountFunction();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
