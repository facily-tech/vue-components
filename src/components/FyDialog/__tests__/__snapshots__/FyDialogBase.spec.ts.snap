// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FyDialogBase.ts should render component and match snapshot 1`] = `
<div class="v-dialog__container">
  <!---->
</div>
`;
